src/Build.UnitTests/Graph/ProjectGraph_Tests.cs

```diff
@@ -1299,6 +1277,19 @@ public void GetTargetsListProjectReferenceTargetsOrDefaultComplexPropagation()
             }
         }
 
+        [Fact]
+        public void ReferencedMultitargetingEntryPointNodeTargetListContainsBuildTarget()
+        {
+            using (var env = TestEnvironment.Create())
+            {
+                TransientTestFile entryProject1 = CreateProjectFile(env, 1, projectReferences: new[] { 2 });
+                TransientTestFile entryProject2 = CreateProjectFile(env, 2, extraContent: MultitargetingSpecificationPropertyGroup);
+                var graph = new ProjectGraph(new HashSet<string> { entryProject1.Path, entryProject2.Path });
+                IReadOnlyDictionary<ProjectGraphNode, ImmutableList<string>> targetLists = graph.GetTargetLists(null);
+                targetLists[key: GetOuterBuild(graph, 2)].ShouldBe(expected: new[] { "Build" });
```

Comment:
Does this list not include `GetTargetFrameworks` (or whatever the test equivalent is)?

I would expect it to merge, so that the outer build of another project can do its select-a-TF process and so the outer build of the referenced multitargeting project can do `Pack`.

Reply:
I initially wanted to make the test case as simple as possible to test the inclusion of `Build` in the referenced node's target list. I've updated the test case to be more robust and generic, verifying the inclusion of the `GetTargetFrameworks` equivalent `AHelperOuter`.